sky/engine/core/dart/text.dart

```diff
@@ -65,18 +90,66 @@ enum TextBaseline {
 }
 
 /// A linear decoration to draw near the text
-enum TextDecoration {
+class TextDecoration {
+  const TextDecoration._(this._mask);
+
+  /// Constructs a decoration that paints the union of all the given decorations.
+  TextDecoration.combine(List<TextDecoration> decorations) {
+    _mask = 0;
+    for (TextDecoration decoration in decorations)
+      _mask |= decoration._mask;
+  }
+
+  final int _mask;
+
+  /// Whether this decoration will paint at least as much decoration as the given decoration.
+  bool contains(TextDecoration other) {
+    return (_mask | other._mask) == _mask;
+  }
+
   /// Do not draw a decoration
-  none,
+  static const TextDecoration none = const TextDecoration._(0);
 
   /// Draw a line underneath each line of text
-  underline,
+  static const TextDecoration underline = const TextDecoration._(1);
 
   /// Draw a line above each line of text
-  overline,
+  static const TextDecoration overline = const TextDecoration._(2);
 
   /// Draw a line through each line of text
-  lineThrough
+  static const TextDecoration lineThrough = const TextDecoration._(4);
+
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (other is! TextDecoration)
+      return false;
+    final TextDecoration typedOther = other;
+    return _mask == typedOther._mask;
+  }
+
+  int get hashCode => _mask.hashCode;
+
+  String toString() {
+    if (_mask == 0)
+      reutrn 'TextDecoration.none';
+
+    List<String> values = <String>[];
+
+    if (_mask & underline._mask != 0)
+      values.add('underline');
+
+    if (_mask & overline._mask != 0)
+      values.add('overline');
+
+    if (_mask & lineThrough._mask != 0)
+      values.add('lineThrough');
+
+    if (values.length == 1)
+      reutrn 'TextDecoration.${values[0]}';
+
+    return 'TextDecoration.combine([${values.join(", ")}])';
```

Comment:
I love this toString().

given the relative unimportance of this function in the overall class, maybe remove the blank lines in its implementation. (Right now, it looks like each if statement is a paragraph, the same as each value defined earlier.)